docs/core/compatibility/core-libraries/9.0/unsafeaccessor-generics.md

```diff
@@ -0,0 +1,51 @@
+---
+title: "Breaking change: Altered UnsafeAccessor support for non-open generics"
+description: Learn about the .NET 9 breaking change in core .NET libraries where support for UnsafeAccessor with non-open generics has been altered.
+ms.date: 07/29/2024
+---
+# Altered UnsafeAccessor support for non-open generics
+
+.NET 8 introduced the <xref:System.Runtime.CompilerServices.UnsafeAccessorAttribute> attribute, which permits access to non-visible members of types (AKA "fast private reflection"). Support for generics in .NET 8 wasn't added due to time constraints. However, in CoreCLR and native AOT, some very narrow and unsupported scenarios involving non-open generic types *did work*. These scenarios should have been blocked, but inadvertently weren't. New restrictions have been added in .NET 9.
+
+For more information and examples, see the [remarks for UnsafeAccessorAttribute](xref:System.Runtime.CompilerServices.UnsafeAccessorAttribute#remarks).
+
+## Previous behavior
+
+In .NET 8, a naive signature look-up on types was implemented, and use of generic types was deemed valid in some cases. For example, the following code succeeded:
+
+```csharp
+[UnsafeAccessor(UnsafeAccessorKind.Method, Name = ".ctor")]
+extern static void CtorAsMethod(List<int> c);
+```
+
+## New behavior
+
+Starting in .NET 9, the fully supported and documented way to consume generic types is to ensure type parameters match type parameters and method parameters match method parameters. These restrictions are necessary because the runtime performs a strict metadata signature match.
```

Comment:
Should this use "argument" and "parameter"?

```suggestion
Starting in .NET 9, the fully supported and documented way to consume generic types is to ensure type arguments match type parameters and method arguments match method parameters. These restrictions are necessary because the runtime performs a strict metadata signature match.
```